KeyVaultEntityRecoverySamples.cs

```diff
@@ -166,50 +148,67 @@ public static async Task DemonstrateBackupAndRestoreAsync()
             // retrieve the vault (or create, if it doesn't exist)
             var vault = await sample.CreateOrRetrieveVaultAsync(rgName, vaultName, enableSoftDelete: false, enablePurgeProtection: false);
             var vaultUri = vault.Properties.VaultUri;
+
+            SecretClient secretClient = sample.GetSecretClient(new Uri(vaultUri));
+
             Console.WriteLine("Operating with vault name '{0}' in resource group '{1}' and location '{2}'", vaultName, rgName, vault.Location);
 
             try
             {
                 // set a secret
                 Console.Write("Setting a new value for secret '{0}'...", secretName);
-                var secretResponse = await sample.DataClient.SetSecretWithHttpMessagesAsync(vaultUri, secretName, Guid.NewGuid().ToString()).ConfigureAwait(false);
+                await secretClient.SetSecretAsync(secretName, Guid.NewGuid().ToString());
                 Console.WriteLine("done.");
 
                 // confirm existence
                 Console.Write("Verifying secret creation...");
-                var retrievedSecretResponse = await sample.DataClient.GetSecretWithHttpMessagesAsync(vaultUri, secretName, secretVersion: String.Empty).ConfigureAwait(false);
+                await secretClient.GetSecretAsync(secretName);
                 Console.WriteLine("done.");
 
                 // backup secret
                 Console.Write("Backing up secret...");
-                var backupResponse = await sample.DataClient.BackupSecretWithHttpMessagesAsync(vaultUri, secretName).ConfigureAwait(false);
+                Response<byte[]> backupResponse = await secretClient.BackupSecretAsync(secretName);
                 Console.WriteLine("done.");
 
                 // delete secret
                 Console.Write("Deleting secret...");
-                await sample.DataClient.DeleteSecretWithHttpMessagesAsync(vaultUri, secretName).ConfigureAwait(false);
+                DeleteSecretOperation deleteSecretOperation = await secretClient.StartDeleteSecretAsync(secretName);
+                // When deleting a secret asynchronously before you purge it, you can await the WaitForCompletionAsync method on the operation
+                await deleteSecretOperation.WaitForCompletionAsync();
                 Console.WriteLine("done.");
 
+                // purge secret
+                Console.Write("Purging deleted secret...");
+                await secretClient.PurgeDeletedSecretAsync(secretName);
+                Console.WriteLine("done.");
+
+                // wait for purge complete
+                Console.Write("Waiting for Purging complete...");
+                Thread.Sleep(60 * 1000);
```

Comment:
While sleeping for a bit is a good idea (and there's nothing more we can do anyway), you should actually have a limited retry loop. Purging can take a while, and will vary (it may be a scheduled task in the backend).

Reply:
@heaths , i am not sure how to implement this limited retry loop. As i think there's nothing more we can do here.